build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/MrjarPlugin.java

```diff
@@ -132,6 +132,44 @@ private void addMrjarSourceset(
         project.getTasks().withType(CheckForbiddenApisTask.class).named(forbiddenApisTaskName).configure(forbiddenApisTask -> {
             forbiddenApisTask.setIgnoreMissingClasses(true);
         });
+
+        return sourceSet;
+    }
+
+    private void createTestTask(Project project, SourceSet sourceSet, int javaVersion) {
+        var jarTask = project.getTasks().withType(Jar.class).named(JavaPlugin.JAR_TASK_NAME);
+        project.getTasks().register(JavaPlugin.TEST_TASK_NAME + javaVersion, Test.class).configure(testTask -> {
```

Comment:
We'll want to have the `check` task depend on this so that they actually get run.